agent/xds/clusters.go

```diff
@@ -533,6 +540,39 @@ func (s *ResourceGenerator) makeGatewayServiceClusters(
 	return clusters, nil
 }
 
+func (s *ResourceGenerator) makeGatewayOutgoingClusterPeeringServiceClusters(cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message, error) {
+	if cfgSnap.Kind != structs.ServiceKindMeshGateway {
+		return nil, fmt.Errorf("unsupported gateway kind %q", cfgSnap.Kind)
+	}
+
+	var clusters []proto.Message
+
+	for _, serviceGroups := range cfgSnap.MeshGateway.PeeringServices {
+		for sn, serviceGroup := range serviceGroups {
+			var clusterName string
+			var isRemote bool
+			if len(serviceGroup) > 0 {
```

Comment:
Same as with endpoints, if we can't get a clusterName we probably shouldn't attempt to make a cluster

Could do:
```go
if len(serviceGroup) == 0 { 
	continue
}
```